python/lsst/daf/butler/registry/queries/_query.py

```diff
@@ -339,6 +562,33 @@ def explain_no_results(
                 yield from followup_query.explain_no_results(db, region=region, followup=False)
                 return
 
+    def order_by(self, db: Database, *args: str) -> None:
+        """Make query return ordered result.
+
+        Parameters
+        ----------
+        db : `Database`
+            Object managing the database connection.
+        args : `str`
+            Names of the columns/dimensions to use for ordering. Column name
+            can be prefixed with minus sign to use descending ordering.
+        """
+        self._order_by = _OrderByHandler(args, self.graph, db)
```

Comment:
I believe that aside from this method, `Query` is immutable and "modifiers" always returns a new object.

Might be expedient to mostly ignore that for now, since I don't think anything depends on that immutabilty (only those `QueryResults` objects use `Query` directly), but it would be good to `return self` and document that that's what user should use, with `self` effectively consumed, in case that's what we want to do later.